### ISSUE TITLE
You have to change the code a little as well. You have to comment out where speaker_id is being referenced in the code. I commented out 3 lines and it works fine for me without further modificaiton;

### ISSUE THREAD

**Issue by gangagyatso4364** [ylacombe/finetune-hf-vits#42]
You have to change the code a little as well. You have to comment out where speaker_id is being referenced in the code. I commented out 3 lines and it works fine for me without further modificaiton;
1-
  model_outputs = model(
                    input_ids=batch["input_ids"],
                    attention_mask=batch["attention_mask"],
                    labels=batch["labels"],
                    labels_attention_mask=batch["labels_attention_mask"],
                    # speaker_id=batch["speaker_id"],
                    return_dict=True,
                    monotonic_alignment_function=maximum_path,
                )
                
2-model_outputs_train = model(
                            input_ids=batch["input_ids"],
                            attention_mask=batch["attention_mask"],
                            labels=batch["labels"],
                            labels_attention_mask=batch["labels_attention_mask"],
                            # speaker_id=batch["speaker_id"],
                            return_dict=True,
                            monotonic_alignment_function=maximum_path,
                        )

3-  model_outputs_train = model(
                        input_ids=batch["input_ids"],
                        attention_mask=batch["attention_mask"],
                        labels=batch["labels"],
                        labels_attention_mask=batch["labels_attention_mask"],
                        # speaker_id=batch["speaker_id"],
                        return_dict=True,
                        monotonic_alignment_function=maximum_path,
                    )

_Originally posted by @muhammadsaadgondal in https://github.com/ylacombe/finetune-hf-vits/issues/35#issuecomment-2282717976_

**Comment by gangagyatso4364** [comment]
Can you share the config file training multiple speakers. ?

**Comment by C0deXG** [comment]
> Can you share the config file training multiple speakers. ?

Just remove all the speaker_id or speaker mentioing in config file so remove all speaker mentioning in the config
